src/api/control/async/asyncMotionProfileController.cpp

```diff
@@ -181,7 +176,7 @@ std::string AsyncMotionProfileController::getPathErrorMessage(const std::vector<
 }
 
 bool AsyncMotionProfileController::removePath(const std::string &ipathId) {
-  if (isRunning.load(std::memory_order_acquire) && getTarget() == ipathId) {
+  if (!isDisabled() && isRunning.load(std::memory_order_acquire) && getTarget() == ipathId) {
```

Comment:
Why force the controller to be enabled? I think removing a path from a disabled controller is fine.

Reply:
This if branch should only activate if a) The controller is enabled (`!isDisabled()`), b) A path is currently running and c) if the path being removed is the currently running path. So it is able to remove any path it wants from the disabled controller.

Reply:
I see, I didn't look at the entire method body.